overlord/snapstate/conflict.go

```diff
@@ -90,6 +90,60 @@ func affectedSnaps(t *state.Task) ([]string, error) {
 	return nil, nil
 }
 
+func checkChangeConflictExclusiveKinds(st *state.State, newExclusiveChangeKind, ignoreChangeID string) error {
+	for _, chg := range st.Changes() {
+		if chg.Status().Ready() {
+			continue
+		}
+		switch chg.Kind() {
+		case "transition-ubuntu-core":
+			return &ChangeConflictError{
+				Message:    "ubuntu-core to core transition in progress, no other changes allowed until this is done",
+				ChangeKind: "transition-ubuntu-core",
+			}
+		case "transition-to-snapd-snap":
+			return &ChangeConflictError{
+				Message:    "transition to snapd snap in progress, no other changes allowed until this is done",
+				ChangeKind: "transition-to-snapd-snap",
+			}
+		case "remodel":
+			if ignoreChangeID != "" && chg.ID() == ignoreChangeID {
+				continue
+			}
+			return &ChangeConflictError{
+				Message:    "remodeling in progress, no other changes allowed until this is done",
+				ChangeKind: "remodel",
+			}
+		case "create-recovery-system":
+			if ignoreChangeID != "" && chg.ID() == ignoreChangeID {
+				continue
+			}
+			return &ChangeConflictError{
+				Message:    "creating recovery system in progress, no other changes allowed until this is done",
+				ChangeKind: "create-recovery-system",
+			}
+		default:
+			if newExclusiveChangeKind != "" {
+				// we want to run a new exclusive change, but other
+				// changes are in progress already
+				msg := fmt.Sprintf("other changes in progress, change %q not allowed until they are done (conflicting change %q)",
```

Comment:
the parentheses could also go before the ",", it's slightly more cohesive that way but no super strong opinion though